CONTRIBUTING.md

```diff
@@ -0,0 +1,112 @@
+Contribute to Che-Theia
+================
+
+There are 3 contribution points:
+- Theia
+- Che-Theia extensions
+- Che-Theia plugins
+
+For each of these kinds of contribution, rebuilding the container image is not needed to code, build and test your contribution.
+We can create a Che workspace that would setup all the containers to build and run Che-Theia. The containers will rely on the same ones used in the CI for the Che-Theia container build. Building che-theia in the che-workspace would be very close to the one build by Docker.
+
+You will find here few devfile that will help you in setting up theses environments
+
+## Devfiles
+Each devfile could be run on any Che instances. Beware that each of them may need a certain amount of memory.
+Devfile could be launched through a factory or [chectl](https://github.com/che-incubator/chectl) cli.
+
+```
+$ chectl workspace:start -f devfiles/che-theia-all.devfile.yaml
+```
+
+## Devfile for contributing to Theia or Che-Theia extensions or Che-plugins
+> Note that this devfile is going to evolve very soon and be splitted into multiple devfiles.
+
+The devfile: [<this.repo>/devfiles/che-theia-all.devfile.yaml](../devfiles/che-theia-all.devfile.yaml)
+
+The goal is to show how to setup a che7 environment to work on che-theia.
+For the workflow we would have a workspace with such containers:
+
+- Theia IDE (Editor) container (a.k.a theia-idexxx). It would have Theia editor with a set of plugins. It will also run the Theia instance for testing our development (theia-dev endpoint).
+- Che-Theia Dev container (a.k.a che-dev). We will build the che-theia (theia + che-theia extensions) and plugins.
+
+All containers have `/projects` folder mounted, which is shared among them.
+
+
+Developer steps:
+
+1. Start the workspace with the devfile, it is cloning Theia and the che-theia plugins.
+2. Setup Theia sources to include Che-theia extension thanks to `che:theia init` command (Che-theia dev container)
+3. ….. [ Code on theia or che-theia extension ]
+4. Build che-theia (theia + che-theia extensions) (Che-theia dev container)
+5. …...[ Code on the a plugin project ]
+6. Build the plugin project (Che-theia dev container)
+7. Run che-theia and will be configured to include the plugin. (Che-theia dev container or theia-ide container)
+
+The following devfile provides examples of commands to build and run che-theia with factory and containers plugins.
+
+### Che-theia init
```

Comment:
I think the title does not explain what this step is about.
maybe "Step1: Checkout all extensions"
And then, I would start by explaining "what we are going to do, why" and then "the how" to do it.